ios/Classes/MapboxMapController.swift

```diff
@@ -810,6 +810,25 @@ class MapboxMapController: NSObject, FlutterPlatformView, MGLMapViewDelegate, Ma
         return trackCameraPosition ? mapView.camera : nil
         
     }
+    /*
+    *  Scan layers from top to bottom and return the first matching feature
+    */
+    private func firstFeatureOnLayers(at: CGPoint) -> MGLFeature? {
+        guard let style = mapView.style else { return  nil}
+       
+        // get layers in order (featureLayerIdentifiers is unordered)
```

Comment:
`featureLayerIdentifiers` is a Set and in Swift set is unordered collection of unique elements https://developer.apple.com/documentation/swift/set

We could also change `featureLayerIdentifiers` to be something else like https://github.com/apple/swift-collections/blob/main/Documentation/OrderedSet.md and then it should just work without any extra methods

Reply:
i though so as well and found ordered set as well.
However i struggled to get it installed as there doesn't seem to be a cocoapods available for it.
Also visibleFeatures expects styleLayerIdentifiers to be a set - so i wonder if i could even replace it?

On the other hand the old approach with featureLayerIdentifiers (even if it would have worked as expected) had the issue that if the user adds the layers out of order with belowLayer, the order of featureLayerIdentifiers would not represent the actual order of the layers on the map anyhow. So we would have to query the layers after we add a layer and recreate featureLayerIdentifiers with the order as they are on the map. This feels just as clunky as this.

Reply:
Does Mapbox maintain somewhere internal ordered list of layers? Maybe use that instead of saving every layer to a Set again?

Reply:
it does in style.layers - but we only want to query the layers added by the user - if not we will get feature tap event even for stuff on the base map